clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -400,12 +412,33 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
             int totalBytes = 0;
             int totalCount = 0;
             FetchResponse response = new FetchResponse(resp.responseBody());
+
+            // Look for OffsetOutOfRange error in fetchResponse
+            Map<TopicPartition, Long> offsetOutOfRangePartitions = new HashMap();
+            for (Map.Entry<TopicPartition, FetchResponse.PartitionData> entry : response.responseData().entrySet()) {
```

Comment:
Because I need to avoid making any change to subscriptions if there is OffsetOutOfRange error.

You are right. Loop below shouldn't need OFFSET_OUT_OF_RANGE any more. I will remove it. Thanks for the catch.